core/src/main/java/org/frankframework/pipes/ForPipe.java

```diff
@@ -0,0 +1,169 @@
+/*
+   Copyright 2024 WeAreFrank!
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package org.frankframework.pipes;
+
+import java.util.Optional;
+
+import lombok.Getter;
+import org.apache.commons.lang3.StringUtils;
+
+import org.frankframework.configuration.ConfigurationException;
+import org.frankframework.core.ParameterException;
+import org.frankframework.core.PipeForward;
+import org.frankframework.core.PipeLineSession;
+import org.frankframework.core.PipeRunException;
+import org.frankframework.core.PipeRunResult;
+import org.frankframework.doc.ElementType;
+import org.frankframework.doc.ElementType.ElementTypes;
+import org.frankframework.doc.Forward;
+import org.frankframework.parameters.ParameterValue;
+import org.frankframework.parameters.ParameterValueList;
+import org.frankframework.stream.Message;
+
+/**
+ * ForPipe is a wrapper to use another pipe a fixed number of times. This can be accomplished by something like:
+ *
+ * <pre>{@code
+ * 		<ForPipe name="forPipe" startAt="0" stopAt="10">
+ * 		 	<Forward name="stop" path="EXIT" />
+ * 		 	<Forward name="continue" path="echoPipe"/>
+ * 		</ForPipe>
+ *
+ * 		<EchoPipe name="echoPipe" getInputFromSessionKey="forPipe.iteration">
+ * 		 	<Forward name="success" path="forPipe"/>
+ * 		</EchoPipe>
+ * }</pre>
+ * <p>
+ * This should call the echoPipe for i=0 until i=10.
+ * <p>
+ *
+ * @author evandongen
+ * @ff.info After completing the for loop, the sessionKey containing the for loop iteration state, will be removed.
+ * @ff.info The default format for the session key is "pipeName-iteration"
+ */
+@Forward(name = "stop", description = "exit for loop")
+@Forward(name = "continue", description = "continue in for loop")
+@ElementType(ElementTypes.ITERATOR)
+public class ForPipe extends FixedForwardPipe {
```

Comment:
I hate to be that guy maar... de FixedForwardPipe heeft standaard een success forward die nu helemaal niet gebruikt wordt 😢

Reply:
Die `success` en `exception` forwards zitten op `AbstractPipe`, dus er zijn meer pipes die hier last van hebben, zoals bijvoorbeeld `XmlIf` en `RegExPipe`. Die forwards naar `FixedForwardPipe` schuiven en alle pipes die een alternatieve flow hebben `AbstractPipe` laten gebruiken is de oplossing.

Reply:
Zo, dat klinkt als een mooi issue voor de 9.0!

Reply:
Ok, ik zal daar een ticket voor maken. Voor nu hoeft er in deze pipe denk ik niets te gebeuren hiervoor?